packages/expect/src/matchers.ts

```diff
@@ -628,10 +628,7 @@ const matchers: MatchersObject = {
       promise: this.promise,
     };
 
-    if (
-      typeof received !== 'string' &&
-      (!received || typeof received.length !== 'number')
-    ) {
+    if (typeof received !== 'string' && typeof received?.length !== 'number') {
```

Comment:
yeah, that guard was probably because `if ('')` evaluates to false. optional chaining on the `length` property should be fine 👍

Reply:
if we don't have a test for `expect('').toHaveLength(0)` we should add one 🙂

Reply:
alright then let me remove the string thing

Reply:
Done

Looks like a test already exists here:
https://github.com/facebook/jest/blob/dcb561b8b7e9151aea3daffc8a377d8cd101c5e5/packages/expect/src/__tests__/matchers.test.js#L1696

Reply:
perfect, thanks for verifying 👍